synapse/http/proxyagent.py

```diff
@@ -263,93 +270,79 @@ def _http_proxy_endpoint(
     reactor: IReactorCore,
     tls_options_factory: IPolicyForHTTPS,
     **kwargs,
-):
+) -> Tuple[Optional[IStreamClientEndpoint], Optional[ProxyCredentials]]:
     """Parses an http proxy setting and returns an endpoint for the proxy
 
     Args:
         proxy: the proxy setting in the form: [scheme://][<username>:<password>@]<host>[:<port>]
-            This currently supports http:// and https:// proxies. 
+            This currently supports http:// and https:// proxies.
             A hostname without scheme is assumed to be http.
 
         reactor: reactor to be used to connect to the proxy
 
         tls_options_factory: the TLS options to use when connecting through a https proxy
+
         kwargs: other args to be passed to HostnameEndpoint
 
     Returns:
-        interfaces.IStreamClientEndpoint|None: endpoint to use to connect to the proxy,
-            or None
-
-    Raises: ValueError if given a proxy with a scheme we don't support.
```

Comment:
We should specify the behaviour of this function when the proxy string is invalid.